src/fmgc/src/guidance/vnav/FlightModel.ts

```diff
@@ -83,7 +88,37 @@ export class FlightModel {
     static getDrag(weight: number, mach: number, delta: number, spdBrkDeflected: boolean, gearExtended: boolean, flapConf: FlapConf): number {
         const Cl = this.getLiftCoefficient(weight, mach, delta);
         const Cd = this.getDragCoefficient(Cl, spdBrkDeflected, gearExtended, flapConf);
-        return 1481.4 * (mach ** 2) * delta * this.wingArea * Cd;
+        const deltaCd = this.getMachCorrection(mach, flapConf);
+
+        return 1481.4 * (mach ** 2) * delta * this.wingArea * (Cd + deltaCd);
+    }
+
+    static getMachCorrection(mach: Mach, flapConf: FlapConf): number {
+        if (flapConf !== FlapConf.CLEAN) {
+            return 0;
+        }
+
+        return this.interpolate(mach, this.machValues, this.dragCoefficientCorrections);
+    }
+
+    /**
+     * Interpolates in a list
+     * @param x The value to look up in in `xs`.
+     * @param xs The table of x values with known y values
+     * @param ys The y values corresponding to the x values in `xs`
+     */
+    static interpolate(x: number, xs: number[], ys: number[]) {
+        if (x <= xs[0]) {
+            return ys[0];
+        }
+
+        for (let i = 0; i < xs.length - 1; i++) {
+            if (x > xs[i] && x < xs[i + 1]) {
```

Comment:
Shouldn’t one of these be or equal?

Reply:
Absolutely right, had just noticed that. 😄